multiapps-mta/src/main/java/org/cloudfoundry/multiapps/mta/handlers/ResourceBatchCalculator.java

```diff
@@ -0,0 +1,56 @@
+package org.cloudfoundry.multiapps.mta.handlers;
+
+import org.apache.commons.collections4.CollectionUtils;
+import org.cloudfoundry.multiapps.mta.handlers.v3.DescriptorHandler;
+import org.cloudfoundry.multiapps.mta.model.DeploymentDescriptor;
+import org.cloudfoundry.multiapps.mta.model.Resource;
+import org.cloudfoundry.multiapps.mta.model.WeightedResource;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.TreeMap;
+
+public class ResourceBatchCalculator {
+
+    private final DeploymentDescriptor deploymentDescriptor;
+    private final DescriptorHandler descriptorHandler;
+
+    public ResourceBatchCalculator(DeploymentDescriptor deploymentDescriptor) {
+        this.deploymentDescriptor = deploymentDescriptor;
+        this.descriptorHandler = new DescriptorHandler();
+
```

Comment:
Move the instantiation in a new method, this way the class can be extended in a test class and the descriptorHandler can be mocked. Are there unit tests for this class